.githooks/pre-commit

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+# stash unstaged changes
+git diff > unstaged.diff
+git apply -R unstaged.diff
+
+# Lint, fix, store exit code, and stage modifications
+rubocop --fail-fast --auto-correct --except Layout/EndOfLine
```

Comment:
We don't care about `Layout/EndOfLine`? 
Is this rubocop using the config in `.rubocop.yml`? If so, can we move it there if we really care? Preferably with a comment why it's needed (windows being annoying I'm guessing).

Reply:
It's some interaction between windows, rubocop, and git core.autocrlf

Reply:
Could we try running rubocop only on changed files? 

https://medium.com/devnetwork/running-rubocop-only-on-modified-files-a21aed86e06d has some ugly commands :P

Reply:
I only want it to run on staged changes, because I often commit parts of a file

Reply:
This is what lines 4, 5, 18 & 19 do

Reply:
Oh wait, I see what you mean... Will do